azure-iot-device/azure/iot/device/provisioning/aio/async_provisioning_device_client.py

```diff
@@ -79,7 +79,11 @@ async def register(self):
 
         register_async = async_adapter.emulate_async(self._pipeline.register)
         register_complete = async_adapter.AwaitableCallback(return_arg_name="result")
-        await register_async(payload=self._provisioning_payload, callback=register_complete)
+        await register_async(
+            payload=self._provisioning_payload,
+            callback=register_complete,
+            client_csr=self.client_csr,
```

Comment:
Should this be an argument on the `register` call that is provided in the operation, or something provided at client creation and stored in the PipelineConfig?

Reply:
it should not be on the config , coz this has nothing to do with the pipeline . This is almost an optional parameter that the user may choose to send via the register call. If the user sends it then there is a certificate in the response. If not proved register can still happen but there will be no certificate in response.

Reply:
Okay, so this value is directly tied to an individual registration then? In that case, why are we putting it as a property on the client?

Reply:
This is more like that payload. Hence followed the pattern. Even the payload has a setter on the client, but is only required while sending resgister.